libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -501,6 +501,26 @@ void BaseDaemon::closeFDs()
     }
 }
 
+namespace {
+/// In debug version on Linux, increase oom score so that clickhouse is killed
+/// first, instead of some service.
+#if defined(__linux__) and not defined(NDEBUG)
+void debug_adjust_oom_score()
+{
+    int file = open("/proc/self/oom_score_adj", O_WRONLY);
+    if (file < 0)
+    {
+        return;
+    }
+    const char score[] = "555";
+    write(file, score, sizeof(score));
```

Comment:
Do we need to write terminating zero to this file?

Reply:
I don't know, it works both ways and the docs say nothing. I changed it to std::string so it doesn't write zero now.